Production Release v.0.0.1-pre+1 (#25)
open-feature/dart-server-sdk: CONTRIBUTING.md

# **Welcome to Open Feature!**

Thank you for contributing to this project. We value your input, and any issues or pull requests adhering to these guidelines are welcome.

## **Code of Conduct**

Please read and follow our [Code of Conduct](https://github.com/open-feature/.github/blob/main/CODE_OF_CONDUCT.md).  
**TL;DR**: Be respectful and professional.

## **Vendor Specific Details**

Vendor specific details are intentionally not included in this module in order to be lightweight and agnostic.
If there are changes needed to enable vendor specific behaviour in code or other extension points, check out [the spec](https://github.com/open-feature/spec).

> Here’s the updated **README** tailored for **Open Feature** and GitHub:

Reviewer: ```suggestion
Here’s the updated **README** tailored for **OpenFeature** and GitHub:

```

Author: @beeme1mr noted.

Reviewer 2: @ABC2015 this is updated cc @beeme1mr

Author: @brian-chebon Noted - resolving this comment.

| `refactor`    | `refactor`             | Code restructuring without functional changes. Example: `refactor(ui): improve layout`. |
| `release`     | `release`              | Prepares a versioned release. Example: `release: v1.2.0`.                      |

---

#### **Best Practices**

1. **Align Commit Type with Branch Type**:
   - A branch like `feat/add-auth-module` should include commits starting with `feat:`.
   - A branch like `fix/login-error` should include commits starting with `fix:`.

2. **Use Scope (Optional)**:
   - Add a scope in parentheses to specify the module or component affected.
   - Examples: `feat(auth)`, `fix(payment)`, `test(ui)`.

3. **Write a Clear Summary**:
   - Use the imperative mood for the summary (e.g., "add", "fix", not "added" or "fixed").

4. **Add Context in the Body (Optional)**:
   - Provide additional details or reasoning for the change.

5. **Reference Issues or PRs in the Footer**:
   - Use the footer for links to related issues or PRs and to indicate breaking changes.
   - Examples:
     ```plaintext
     Fixes #123
     BREAKING CHANGE: Deprecated legacy login endpoints.
     ```

---

> ### **Branching Rules and Protection**

Reviewer: Reminder to myself to check if these protections are actually in place.

Here’s your updated **Submitting a Pull Request** section with improved formatting and alignment for clarity and readability:

## **Submitting a Pull Request**

Contributing to the project is highly encouraged! Please follow these steps to submit a well-structured pull request (PR):

### **Steps to Submit a Pull Request**

1. **Create a New Branch**:
   - Use the appropriate branch naming convention:
     ```bash
     git checkout -b feat/new-feature
     ```
   - Ensure your branch is based on the latest `development` branch:
     ```bash
     git fetch origin
     git checkout development
     git rebase origin/development
     ```

2. **Implement Changes**:
   - Write clean, modular code following the project's coding standards.
   - Add appropriate unit and integration tests to ensure coverage.

3. **Run Tests Locally**:
   - Validate your changes locally to ensure they don't break existing functionality:
     ```bash
     dart analyze
     dart test
     ```

4. **Commit Your Changes**:
   - Use meaningful commit messages following the [Conventional Commits](https://www.conventionalcommits.org) format:
     ```bash

>      git commit -s -m "feat(sdk): implement new feature"

Reviewer: ```suggestion
     git commit -s -m "feat: implement new feature"
```

I don't think `sdk` is required in this context since we're not working in a monorepo. At least not yet.

Author: @beeme1mr makes sense. cc: @brian-chebon

Reviewer 2: @ABC2015 updated cc @beeme1mr

---

### **Best Practices for Pull Requests**

- **Keep Pull Requests Small**:
  - Submit changes in manageable chunks to make reviews easier.
- **Follow Coding Standards**:
  - Adhere to the project's coding style and guidelines.
- **Document Changes**:
  - Update relevant documentation, if applicable, to reflect your changes.
- **Add Tests**:
  - Ensure new features or bug fixes are thoroughly tested.
- **Reference Issues**:
  - Use `Fixes #<issue-number>` or `Closes #<issue-number>` in the PR description to auto-close related issues.

---

### **Example Workflow**

```bash
# Create a new branch
git checkout -b feat/add-auth

# Implement changes
# ...

# Run tests
dart analyze
dart test

# Commit changes
git commit -m "feat(auth): add OAuth 2.0 authentication"

# Push changes
git push origin feat/add-auth

# Open a pull request on GitHub
# Provide a clear description linking the related issue (e.g., Fixes #45)
```

## **Versioning Standards**

We follow Dart's [semantic versioning](https://semver.org) conventions:
- `x.y.z+1`: For patch releases during pre-1.0 development.
- `x.y.z`: For stable, production-ready versions.

## **Contacting Us**

> - Join our regular meetings [here](https://github.com/open-feature/community/#meetings-and-events).

Reviewer: ```suggestion
- Join our regular meetings [here](https://openfeature.dev/community/#community-meetings).

```

Author: @beeme1mr thanks for catching that - will update:  cc: @brian-chebon

Reviewer 2: @ABC2015 this is done cc @beeme1mr

Author: @brian-chebon Noted - resolving this comment.
